### PR TITLE
[ share-repo ] can share repo to any groups

### PR DESCRIPTION
**Pull request by zMingGit**

### CODE REVIEW

File: seahub/api2/endpoints/all_groups.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+from seahub.api2.authentication import TokenAuthentication
+from seahub.api2.throttling import UserRateThrottle
+from seahub.api2.endpoints.groups import get_group_info
+from seahub.avatar.settings import GROUP_AVATAR_DEFAULT_SIZE
+from seahub.utils import is_org_context
+from constance import config
+
+
+class AllGroupsView(APIView):
+    """Only for share group selection"""
```
Comment (xiez) [inline seahub/api2/endpoints/all_groups.py]: List all groups

File: seahub/api2/endpoints/all_groups.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+from seahub.api2.authentication import TokenAuthentication
+from seahub.api2.throttling import UserRateThrottle
+from seahub.api2.endpoints.groups import get_group_info
+from seahub.avatar.settings import GROUP_AVATAR_DEFAULT_SIZE
+from seahub.utils import is_org_context
+from constance import config
+
+
+class AllGroupsView(APIView):
+    """List all groups"""
```
Comment (llj) [inline seahub/api2/endpoints/all_groups.py]: 注释写这里？位置不对吧

File: seahub/api2/endpoints/all_groups.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    authentication_classes = (TokenAuthentication, SessionAuthentication)
+    permission_classes = (IsAuthenticated,)
+    throttle_classes = (UserRateThrottle, )
+
+    def get(self, request):
+        """Get personal groups or get all groups
```
Comment (llj) [inline seahub/api2/endpoints/all_groups.py]: which one?

File: seahub/api2/endpoints/all_groups.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        """
+        org_id = None
+        username = request.user.username
```
Comment (llj) [inline seahub/api2/endpoints/all_groups.py]: `org_id`, `username` 后面都没用到。

**Pull request merged**